### PR TITLE
Imporve Z-Order with Spark3.3

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

We can inject rebalance before Z-Order to avoid data skew.

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]
Thanks for pinging me, this PR looks good, I learned from it.

File: extensions/spark/kyuubi-extension-spark-3-3/src/main/scala/org/apache/kyuubi/sql/zorder/InsertZorderBeforeWriting33.scala [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+  def insertZorder(
+      catalogTable: CatalogTable,
+      plan: LogicalPlan,
+      dynamicPartitionColumns: Seq[Attribute]): LogicalPlan = {
+    if (!canInsertZorder(plan)) {
+      return plan
+    }
+    val cols = getZorderColumns(catalogTable.properties)
+    val attrs = plan.output.map(attr => (attr.name, attr)).toMap
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-extension-spark-3-3/src/main/scala/org/apache/kyuubi/sql/zorder/InsertZorderBeforeWriting33.scala:66]: shall we respect spark's resolver?

Comment (ulysses-you) [reply extensions/spark/kyuubi-extension-spark-3-3/src/main/scala/org/apache/kyuubi/sql/zorder/InsertZorderBeforeWriting33.scala]: sure removed

**Review by yaooqinn — APPROVED** [review]

**Review by cfmcgrady — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**